tfworker/providers/base.py

```diff
@@ -62,6 +74,38 @@ def clean(self, deployment, limit, config):
         """Nothing to do here so far"""
         pass
 
+    def _hclify(self, s, depth=4):
+        """
+        _hcify is a recursive function that takes a string, list or dict
+        and turns the results into an HCL compliant string.
+        """
+        space = ' '
+        result = []
+        if isinstance(s, str):
+            tmps = s.replace('"', "").replace("'", "")
+            result.append(f"{space * depth}{s}")
+        elif isinstance(s, list):
+            tmps = [i.replace('"', "").replace("'", "") for i in s]
+            result.append(f"{space * depth}{s}")
+        elif isinstance(s, dict):
+            # unfortunately, HCL doesn't allow for keys to be quoted so a further
+            # check of the value is required to determine how to handle the key
+            for k in s.keys():
+                if isinstance(s[k], str):
+                    result.append(f"{space * depth}{k} = \"{s[k]}\"")
+                elif isinstance(s[k], list):
+                    result.append(f"{space * depth}{k} = [{s[k]}]")
+                elif isinstance(s[k], dict):
+                    # decrease depth by 4 to account for extra depth added by hclyifying the key
+                    result.append(f"{space * (depth-4)}{self._hclify(k, depth=depth)} = {{")
+                    result.append(self._hclify(s[k], depth=depth + 2))
+                    result.append(f"{space * depth}}}")
+                else:
+                    raise TypeError(f"Expected string, list or dict, got {type(s[k])}")
```

Comment:
The only thing I might add is some context so that the user knows which setting has the wrong type. That may be indicated somewhere else in the output, so just mentioning in case it's a good suggestion.

Reply:
It should be technically impossible to end up with the wrong type since the inputs are from the YAML, and if the type is unknown other than spitting out the __str__ representation of the entire object (which could be problematic) there's no way for this function to know what to call it.